synapse/handlers/room_batch.py

```diff
@@ -281,15 +297,22 @@ async def persist_historical_events(
             room_id: Room where you want the events persisted in.
             inherited_depth: The depth to create the events at (you will
                 probably by calling inherit_depth_from_prev_ids(...)).
-            auth_event_ids: Define which events allow you to create the given
-                event in the room.
+            initial_state_event_ids:
+                This is used to set explicit state for the insertion event at
+                the start of the historical batch since it's floating with no
+                prev_events to derive state from automatically.
             app_service_requester: The requester of an application service.
 
         Returns:
             List of persisted event IDs
         """
         assert app_service_requester.app_service
 
+        # We expect the first event in a historical batch to be an insertion event
+        assert events_to_create[0]["type"] == EventTypes.MSC2716_INSERTION
+        # We expect the last event in a historical batch to be an batch event
+        assert events_to_create[-1]["type"] == EventTypes.MSC2716_BATCH
```

Comment:
Do we need to be any more careful about undefined index errors here? We expect the `persist_historical_events` usage to pass in at least a `insertion` and `batch` event which `handle_batch_of_events` does. We want things to explode if the usage changes to something unexpected.

Reply:
I think this is fine, it'll explode quickly if dodge things happen which is all we really need.